server/src/main/java/com/distkv/server/storeserver/runtime/StoreRuntime.java

```diff
@@ -28,34 +30,27 @@
    */
   private ExpirationManager expirationManager;
 
+  private HeartbeatManager heartbeatManager;
+
   private WorkerPool workerPool;
 
-  private List<SlaveClient> slaveClients;
+  private volatile CopyOnWriteArrayList<SlaveClient> slaveClients;
+
+  private volatile NodeInfo nodeInfo;
 
   public StoreRuntime(StoreConfig config) {
     this.config = config;
     storeEngine = new KVStoreImpl();
     expirationManager = new ExpirationManager(config);
-
-    if (config.isMaster()) {
-      slaveClients = new ArrayList<>();
-      for (String slaveAddr : config.getSlaveAddresses()) {
-        final SlaveClient[] client = {null};
-        RuntimeUtil.waitForCondition(() -> {
-          try {
-            client[0] = new SlaveClient(slaveAddr);
-            return true;
-            //TODO : Drpc need to add a Exception to cover
-            // io.netty.channel.AbstractChannel$AnnotatedConnectException
-          } catch (Exception e) {
-            return false;
-          }
-        }, 5 * 60 * 1000);
-        slaveClients.add(client[0]);
-        LOGGER.info("Connecting to slave(" + slaveAddr + ") success");
-      }
-    } else {
-      slaveClients = null;
+    slaveClients = new CopyOnWriteArrayList<>();
+    nodeInfo = new NodeInfo(false, config.getNodeId(),
+        String.format("distkv://%s:%d", config.getIp(), config.getPort()));
+    if (config.getMode().equals("distributed")) {
+      heartbeatManager = new HeartbeatManager(
```

Comment:
I perfer the hearbeat manager has same level of abstractions as store runtime. how do you think?